integrations/fluid.py

```diff
@@ -2,57 +2,74 @@
 from constants.integration_ids import IntegrationID
 from models.integration import Integration
 from utils.web3_utils import call_with_retry, W3_BY_CHAIN
-from utils.fluid import vaultResolver_contract, vaultFactory_contract
+from utils.fluid import vaultResolver_contract, vaultFactory_contract, vaultPositionResolver_contract
+from constants.fluid import susde
+import json
 
 class FluidIntegration(
     Integration
-):  # TODO: Change FluidIntegration to the name of the protocol
+):
     def __init__(self):
         super().__init__(
-            IntegrationID.FLUID,  # TODO: Change EXAMPLE to the name of the protocol
-            21016131,  # TODO: Change 20000000 to the start block of the protocol when events/balances should first be tracked
-            Chain.ETHEREUM,  # TODO: Change ETHEREUM to the chain of the protocol
-            None,  # TODO: Optional, change None to a list of SummaryColumn enums that this protocol should be associated with, see Pendle grouping example
-            20,  # TODO: Change 20 to the sats multiplier for the protocol that has been agreed upon
-            1,  # TODO: Almost always 1, optionally change to a different value if an adjustment needs to be applied to balances
-            None,  # TODO: Optional, change None to the end block of the protocol when events/balances should stop being tracked
-            None,  # TODO: Optional, change None to a function that takes a block number and returns the reward multiplier for that block if it has or will change over time
+            IntegrationID.FLUID,
+            21016131,
+            Chain.ETHEREUM,
+            None,
+            5,
+            1,
+            None,
+            None,
         )
 
-    # TODO: Implement function for getting a given user's balance at a given block
     def get_balance(self, user: str, block: int) -> float:
         balance = 0
         try:
-            userPositions, _ = call_with_retry(vaultResolver_contract.functions.positionsByUser(user), block)
-            print(userPositions)
+            userPositions, vaultEntireDatas = call_with_retry(vaultResolver_contract.functions.positionsByUser(user), block)
             for userPosition in userPositions:
+                if vaultEntireDatas[0][3][8][0] != susde:
+                    continue
                 balance += userPosition[9]
             return balance/1e18
         except Exception as e:
             return 0
 
 
-    # TODO: Implement function for getting all participants of the protocol, ever
-    # Important: This function should only be called once and should cache the results by setting self.participants
     def get_participants(self) -> list:
         participants = []
         current_block = W3_BY_CHAIN[self.chain]["w3"].eth.get_block_number()
+
+        relevant_vaults = self.get_relevant_vaults(current_block)
+        relavantNfts = []
+
+        for vault in relevant_vaults:
```

Comment:
why not for all relevant_vaults, do 
`getAllVaultPositions` at the VaultPositionsResolver which already includes owner? reduces RPC calls significantly? 

> use the VaultPositionsResolver getAllVaultPositions(vault) and map to owner in returned UserPosition struct

Reply:
made the changes